config/config-api/src/main/java/com/thoughtworks/go/config/migration/UrlDenormalizerXSLTMigration120.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config.migration;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class UrlDenormalizerXSLTMigration120 {
+
+    public static String urlWithoutCredentials(String originalUrl) {
+        try {
+            if (isSupportedUrl(originalUrl)) {
+                String[] credentials = getCredentials(originalUrl);
+                if (credentials != null) {
+
+                    URI url = new URI(originalUrl);
+                    return new URI(url.getScheme(), null, url.getHost(), url.getPort(), url.getPath(), url.getQuery(), url.getFragment()).toString();
+                }
+            }
+            return originalUrl;
+        } catch (URISyntaxException e) {
+            return originalUrl;
+        }
+    }
+
+    public static String getUsername(String originalUrl) {
+        try {
+            if (isSupportedUrl(originalUrl)) {
+                String[] credentials = getCredentials(originalUrl);
+                if (credentials != null) {
```

Comment:
If `getCredentials(originalUrl)` returns one argument then we are considering it as `username`. It will be displayed in plain text in API calls.